packaging/build_rpm.py

```diff
@@ -0,0 +1,375 @@
+        if f.endswith('.x86_64.rpm') or f.endswith('.noarch.rpm'):
+            final_rpm = path_join(RESULT_DIR, f)
+            break
+    else:
+        raise RuntimeError('final rpm not found')
+
+    return final_rpm
+
+
+def get_rpmbuild_defines():
+    """
+    Return a dictionary of macros from `version_info`
+    """
+    defines = {}
+    with open(path_join(SCRIPT_DIR, 'version_info')) as f:
+        for line in f:
+            line = line.strip()
+            if line and not line.startswith('#'):
+                k, v = line.split(' ', 1)
+                defines[k] = v
+
+    return defines
+
+
+def render_spec_file(spec_file, defines):
+    """
+    Apply the defines to the spec file. We don't use `--define` because
+    `mock`'s `--rpmbuild-opts` doesn't apply to `--buildsrpm`
+
+    Returns the path to the rendered spec file
+    """
+    with open(spec_file) as f:
+        spec_content = f.read()
+
+    for k, v in defines.items():
+        spec_content = spec_content.replace('%{' + k + '}', v)
+
+    spec_file_out = spec_file.replace('.spec', '.rendered.spec')
+    with open(spec_file_out, 'w') as f:
+        f.write(spec_content)
+
+    return spec_file_out
+
+
+def run_vagrant(cmd, *args, **kwargs):
+    """Run command on Vagrant box"""
+    func = kwargs.pop('_func', check_call)
+
+    logger.info('running <{cmd}> on vagrant builder'.format(cmd=cmd))
+    return func(
+            ['vagrant', 'ssh', 'builder', '-c', cmd],
+            *args, **kwargs)
+
+
+def get_dependencies(spec_file):
+    """get the list of dependencies"""
+    dependencies_file = DEPENDENCIES_FILE.format(spec_file=spec_file)
+
+    try:
+        with open(dependencies_file) as f:
+            dependencies = f.readlines()
+    except IOError:
+        logger.info(
+                'no dependencies file found. skipping dependencies install '
+                '(this is only required for internal cloudify dependencies)')
+        dependencies = []
+
+    return dependencies
+
+
+def install_dependencies(spec_file, source):
+    """
+    Check for and install any internal dependencies.
+
+    Place dependencies, one package name per line, in a file called
+    '<spec_file_name>.dependencies'
+
+    e.g.:
+        $ cat cloudify-premium/packages/cloudify-premium.spec.dependencies
+        cloudify-rest-service
+    """
+    dependencies = get_dependencies(spec_file)
+
+    if not dependencies:
+        return
+
+    logger.info(
+            'installing dependencies from <{file}>. RPMs dir: {dir}'.format(
+                file=DEPENDENCIES_FILE.format(spec_file=spec_file),
+                dir=RPMS_DIR,
+                ))
+
+    rpms = {}
+    for f in listdir(RPMS_DIR):
+        f = f.strip()
+        if any(f.endswith(y) for y in ('.x86_64.rpm', '.noarch.rpm')):
+            name = check_output([
+                'rpm', '-qp', '--queryformat', '%{NAME}', f,
+                ])
+            rpms[name] = f
+
+    logger.info(('found RPMs:', rpms))
```

Comment:
Why the tuple in here? :)

Reply:
logger wasn't behaving how I thought it would with just 2 arguments. Perhaps this should be updated to present the list of RPMs more neatly anyway

Reply:
`logger.info('found RPMs: %s', rpms)` is the right syntax

Reply:
Replaced with a log line for each found RPM instead. The printed dict looked ugly anyway